Lambdas/GHHooks/SecretsRetriever.swift

```diff
@@ -23,6 +41,30 @@ struct SecretsRetriever {
         guard let secret = secret.secretString else {
             throw Errors.secretNotFound(arn: arn)
         }
-        return secret
+        return Secret(secret)
+    }
+
+    private func withLock<T>(block: () async throws -> T) async rethrows -> T {
+        if isLoading {
+            await withCheckedContinuation { continuation in
+                loadWaiters.append(continuation)
+            }
+        }
+
+        isLoading = true
+        defer { isLoading = false }
+
+        let value = try await block()
```

Comment:
You allow a thrown error from here to exit the method without resuming any waiters, that'll deadlock anything that ended up in a wait during the failed load.